website/content/en/docs/kubebuilder/references/client.md

```diff
@@ -0,0 +1,574 @@
+// object's status subresource
+func (c Client) Status() (client.StatusWriter, error)
+```
+
+Example:
+```Go
+import (
+	"context"
+	cachev1alpha1 "github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	ctx := context.TODO()
+	mem := &cachev1alpha1.Memcached{}
+	err := r.Get(ctx, request.NamespacedName, mem)
+
+	...
+
+	// Update
+	mem.Status.Nodes = []string{"pod1", "pod2"}
+	err := r.Status().Update(ctx, mem)
+
+	...
+
+	// Patch
+	patch := client.MergeFrom(mem.DeepCopy())
+	mem.Status.Nodes = []string{"pod1", "pod2", "pod3"}
+	err := r.Status().Patch(ctx, mem, patch)
+
+	...
+}
+```
+
+[status-writer]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#StatusWriter
+
+#### Delete
+
+```Go
+// Delete deletes the given obj from Kubernetes cluster.
+func (c Client) Delete(ctx context.Context, obj runtime.Object, opts ...client.DeleteOption) error
+```
+
+A `client.DeleteOption` is an interface that sets [`client.DeleteOptions`][delete-opts] fields. A `client.DeleteOption` is created by using one of the provided implementations: [`GracePeriodSeconds`][grace-period-seconds], [`Preconditions`][preconditions], [`PropagationPolicy`][propagation-policy].
+
+Example:
+
+```Go
+import (
+	"context"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	...
+
+	pod := &v1.Pod{}
+	err := r.Get(context.TODO(), request.NamespacedName, pod)
+
+	...
+
+	ctx := context.TODO()
+	if pod.Status.Phase == v1.PodUnknown {
+		// Delete the pod after 5 seconds.
+		err := r.Delete(ctx, pod, client.GracePeriodSeconds(5))
+		...
+	}
+
+	...
+}
+```
+
+[delete-opts]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#DeleteOptions
+[grace-period-seconds]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#GracePeriodSeconds
+[preconditions]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#Preconditions
+[propagation-policy]:https://godoc.org/sigs.k8s.io/controller-runtime/pkg/client#PropagationPolicy
+
+#### DeleteAllOf
+
+```Go
+// DeleteAllOf deletes all objects of the given type matching the given options.
+func (c Client) DeleteAllOf(ctx context.Context, obj runtime.Object, opts ...client.DeleteAllOfOption) error
+```
+
+A `client.DeleteAllOfOption` is an interface that sets [`client.DeleteAllOfOptions`][deleteallof-opts] fields. A `client.DeleteAllOfOption` wraps a [`client.ListOption`](#list) and [`client.DeleteOption`](#delete).
+
+Example:
+
+```Go
+import (
+	"context"
+	"fmt"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *ReconcileApp) Reconcile(request reconcile.Request) (reconcile.Result, error) {
```

Comment:
Update for new scaffold. `request reconcile.Request`==>`req ctrl.Request`
Update the imports as well `"sigs.k8s.io/controller-runtime/pkg/reconcile"`==>`ctrl "sigs.k8s.io/controller-runtime"`